instrumentation/log4j/log4j-context-data/log4j-context-data-common/testing/src/main/java/io/opentelemetry/instrumentation/log4j/contextdata/Log4j2Test.java

```diff
@@ -132,4 +133,31 @@ void testIdsWhenSpan() {
     assertThat(events.get(3).getContextData().get(getLoggingKey("trace_flags"))).isEqualTo("01");
     assertThat(events.get(3).getContextData().get("baggage.baggage_key")).isNull();
   }
+
+  @Test
+  void testNoOverrideTraceId() {
+    Logger logger = LogManager.getLogger("TestLogger");
+
+    AtomicReference<Span> spanParent = new AtomicReference<>();
+    ThreadContext.put(getLoggingKey("trace_id"), "test_traceId");
+    ThreadContext.put(getLoggingKey("span_id"), "test_spanId");
+    ThreadContext.put(getLoggingKey("trace_flags"), "test_traceFlag");
+    getInstrumentationExtension()
+        .runWithSpan(
+            "test",
+            () -> {
+              spanParent.set(Span.current());
+              logger.info("log span parent");
+            });
+    List<ListAppender.LoggedEvent> events = ListAppender.get().getEvents();
+    assertThat(events.size()).isEqualTo(1);
+    assertThat(events.get(0).getMessage()).isEqualTo("log span parent");
+    assertThat(events.get(0).getContextData().get(getLoggingKey("trace_id")))
+        .isEqualTo("test_traceId");
+    assertThat(events.get(0).getContextData().get(getLoggingKey("span_id")))
+        .isEqualTo("test_spanId");
+    assertThat(events.get(0).getContextData().get(getLoggingKey("trace_flags")))
+        .isEqualTo("test_traceFlag");
+    ThreadContext.clearAll();
```

Comment:
Imo it would be better to move this before assertions. Currently if an assertion were to fail missing this cleanup will break other tests too. Could also consider placing it in a finally block or using our AutoCleanupExtension.